### DISCUSSION TITLE
Using li2018 auto scheduler: Error: Atomic updates are not supported inside Metal kernels

### DISCUSSION THREAD

**Discussion by vakokako** [halide/Halide#6031] [Q&A]
I'm using `li2018` autoscheduler for gpu target. I get the following exceptions on targets `x86-64-osx-avx-f16c-metal-sse41` and  `x86-64-windows-avx-avx2-d3d12compute-f16c-fma-sse41`:
```
Error: Atomic updates are not supported inside Metal kernels
Error: Atomics operations are not supported inside D3D12Compute kernel.
```
`OpenCl` and `Cuda` are working correctly without exception.

**Comment by alexreinking** [comment]
The Li2018 autoscheduler tries to generate schedules that use `atomic`, but that scheduling directive isn't implemented for d3d12 or metal. I'm not sure if anyone is working on that, but the error message does at least tell you that this scenario is not (yet) supported.
